jupyter_ydoc/ydoc.py

```diff
@@ -76,22 +76,30 @@ def observe(self, callback):
 class YNotebook(YBaseDoc):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._ycells = self._ydoc.get_array("cells")
         self._ymeta = self._ydoc.get_map("meta")
+        self._ycells = self._ydoc.get_array("cells")
+        self._ymetadata = self._ydoc.get_map("metadata")
 
     @property
     def source(self):
-        cells = self._ycells.to_json()
         meta = self._ymeta.to_json()
-        cast_all(cells, float, int)
+        cells = self._ycells.to_json()
+        metadata = self._ymetadata.to_json()
         cast_all(meta, float, int)
+        cast_all(cells, float, int)
+        cast_all(metadata, float, int)
         for cell in cells:
             if "id" in cell and meta["nbformat"] == 4 and meta["nbformat_minor"] <= 4:
                 # strip cell IDs if we have notebook format 4.0-4.4
                 del cell["id"]
+            if (cell["cell_type"] == "raw" or cell["cell_type"] == "raw") and len(
```

Comment:
There is the same condition twice: `cell["cell_type"] == "raw" or cell["cell_type"] == "raw"`.
Also, you might want to use `not cell["attachments"]` instead of `len(cell["attachments"]) == 0`.